# Project to simulate a data set using numpy.random

#### The real world phenomenom I have chosen for this project is the gender breakdown at senior management level in Institutes of Technology.   
I will look at the following variables:   
Age   
Qualification   
Years of Experience   
Grade   
Gender      

The age variable will be a non-negative real number between 25 & 66.
The Qualification variable will be a categorical variable with one of 3 different values: Level 8, Level 9, Level 10.
The years of experience variable will be a non-negative real number, and will be based on age and qualification.
The Grade variable will be a categorical variable with one of 7 different values: Assistant Lecturer, Lecturer, Senior Lecturer 1, Senior Lecturer 2, Senior Lecturer 3, Registrar, President
The Gender variable will be either male or female
The determinant of whether a staff member has reached senior management level will be their Grade - those at SL2, SL3, Registrar & President will be considered at Senior Management level
Only staff categorised as Academic will be considered in this project.   


In [2]:
#Import pandas, create columns for dataset.
import pandas as pd
import numpy as np


In [6]:
#Insert age, gender & qualification columns to dataframe
gender = ['male', 'female']
grade = ['Assistant Lecturer', 'Lecturer', 'Senior Lecturer 1', 'Senior Lecturer 2', 'Senior Lecturer 3', 'Registrar', 'President']
df = pd.DataFrame({'Age': np.random.randint(25,66,450),
                   'Gender': np.random.choice(gender, 450,p=[0.45, 0.55]), 'Grade': np.random.choice(grade, 450, p=[.1573, .7584, .0225, .0393, .0169, .0028, .0028])})
#Function to assign qual depending on age, i.e. <27 cannot hold PhD
def qual(row):

                if row['Age']>= 27:

                    return np.random.randint (8,11)
                
                else: 
                    
                    return np.random.randint (8,9)
                                  
qual =df.apply (lambda row: qual(row),axis=1)
df['Qualification'] = qual
df


Age  Gender               Grade  Qualification
0     33    male            Lecturer              8
1     27  female  Assistant Lecturer              8
2     58  female            Lecturer             10
3     61  female            Lecturer             10
4     55    male            Lecturer             10
5     53    male            Lecturer              8
6     60    male            Lecturer              9
7     56  female            Lecturer              9
8     37    male            Lecturer             10
9     55    male  Assistant Lecturer             10
10    58  female            Lecturer             10
11    26  female            Lecturer              8
12    27  female            Lecturer              8
13    34  female            Lecturer              8
14    38  female            Lecturer             10
15    40    male            Lecturer              8
16    27    male            Lecturer              9
17    52    male            Lecturer              8
18    58  female  Assistant Lecturer             10
19    29  female            Lecturer             10
20    59    male            Lecturer              8
21    30    male            Lecturer              8
22    37  female            Lecturer              9
23    40    male            Lecturer              8
24    41  female  Assistant Lecturer             10
25    65  female            Lecturer              9
26    27    male            Lecturer              8
27    29    male            Lecturer             10
28    26  female            Lecturer              8
29    41  female  Assistant Lecturer             10
..   ...     ...                 ...            ...
420   63    male            Lecturer              9
421   36    male            Lecturer              9
422   41  female            Lecturer              8
423   37    male            Lecturer              8
424   58  female            Lecturer             10
425   63  female            Lecturer              9
426   61    male            Lecturer             10
427   49    male            Lecturer             10
428   62  female  Assistant Lecturer             10
429   41  female  Assistant Lecturer             10
430   48    male   Senior Lecturer 3              8
431   26    male            Lecturer              8
432   61  female  Assistant Lecturer              8
433   26  female  Assistant Lecturer              8
434   55  female  Assistant Lecturer             10
435   25  female            Lecturer              8
436   59  female            Lecturer              8
437   59  female            Lecturer              8
438   45  female            Lecturer              8
439   61  female            Lecturer              9
440   64  female            Lecturer              9
441   28    male            Lecturer              8
442   25  female            Lecturer              8
443   57  female            Lecturer              9
444   60  female            Lecturer             10
445   46    male            Lecturer              8
446   53    male            Lecturer              8
447   61  female  Assistant Lecturer              8
448   35    male  Assistant Lecturer              9
449   51  female  Assistant Lecturer              8

[450 rows x 4 columns]

In [4]:
#Checking that all quals are showing in dataframe
df['Qualification'].unique()

array([10,  8,  9], dtype=int64)

After some research into the GMIT staff data I discovered that currently 21% of academic staff hold a PhD, 62% hold a Master's Degree, and the remainder an Honours Degree. I also found that the gender breakdown in academic staff is 40% male, 60% female. After running the above code I realised that I need to look at the relationship between age & qualification, as, for example, a 25 year old staff member is unlikely to hold a PhD, given that the acadmic minimum requirements are a qualification to at least Level 8 plus three years experience. A Level 8 qualification generally takes 3/4 years, a Masters degree another one, and a PhD at least another 3/4, so the minimum age of a PhD holder employed by GMIT in an academic role cannot be less than 27. I couldn't find any recent data on the average age of completion of a PhD, however the median age of a PhD student in Ireland was 26.5 years in 2013, and in 1997 the typical age of a PhD graduate in Ireland was between 24 & 27. 

### References
https://nonpartisaneducation.org/Review/Resources/Int__lHigherEd_AppendixA.pdf   
http://www.iua.ie/wp-content/uploads/2013/08/IUS-01-2013-Report-on-PhD-Students.pdf   
http://hea.ie/assets/uploads/2018/01/Higher-Education-Institutional-Staff-Profiles-by-Gender-2018.pdf
